src/Carbon/PHPStan/MacroMethodReflection.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of the Carbon package.
+ *
+ * (c) Brian Nesbitt <brian@nesbot.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Carbon\PHPStan;
+
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\ParametersAcceptor;
+use PHPStan\TrinaryLogic;
+use PHPStan\Type\Type;
+
+class MacroMethodReflection implements MethodReflection
+{
+    /** @var ClassReflection */
+    private $declaringClass;
+
+    /** @var string */
+    private $methodName;
+
+    /** @var ParametersAcceptor */
+    private $macroClosureType;
+
+    /** @var bool */
+    private $static;
+
+    /** @var string|null */
+    private $docComment;
```

Comment:
If you prefer, we dropped PHP < 8.1 on master branch:
```suggestion
    private ClassReflection $declaringClass;
    private string $methodName;
    private ParametersAcceptor $macroClosureType;
    private bool $static;
    private ?string $docComment;
```

Reply:
Yeah, sure, I noticed that but on the other hand the old code still used PHPDocs so I thought there was a reason for that.